process/block/sovereign/incomingHeader/incomingEventsProc/eventProcScCall_test.go

```diff
@@ -0,0 +1,87 @@
+package incomingEventsProc
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	"github.com/stretchr/testify/require"
+
+	errorsMx "github.com/multiversx/mx-chain-go/errors"
+)
+
+func TestNewEventProcScCall(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil marshaller, should return error", func(t *testing.T) {
```

Comment:
There is no need to test all these if branches, since you use a common func `checkArgs`
We should refactor these tests to only test `checkArgs` for all paths and only check one error path in constructors that use that function